src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -199,92 +126,40 @@ public CommentAttributes getComment(CommentAttributes commentToGet) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, commentViewerType);
         
         List<Comment> comments = getCommentEntitiesForCommentViewer(courseId, commentViewerType);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return getAttributesListFromEntitiesList(comments);
     }
     
-    /*
-     * Get comment for the sending state (SENT|SENDING|PENDING)
-     */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForSendingState(String courseId, CommentSendingState state) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForSendingState(courseId, state);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForSendingState(courseId, "PLACEHOLDER", state);
     }
     
     /*
      * Get comments for a course
      */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForCourse(String courseId) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForCourse(courseId);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForCourse(courseId);
     }
     
-    
-    /*
-     * Update comment from old state to new state
-     */
     public void updateComments(String courseId, CommentSendingState oldState, CommentSendingState newState) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForSendingState(courseId, oldState);
-        
-        for (Comment comment : comments) {
-            comment.setSendingState(newState);
-        }
-        
-        log.info(Const.SystemParams.COURSE_BACKUP_LOG_MSG + courseId);
-        getPM().close();
+        super.updateComments(courseId, "PLACEHOLDER", oldState, newState);
```

Comment:
Can rename the string PLACEHOLDER to something meaningful? and turn it into a constant.
it's a useless param for comment, but important for frc. but newcomers may think confused